interactive_map_manager:
  ros__parameters:
    namespace: "atlas"
    use_sim_time: true
    #pcd_map_filename: /home/atas/colcon_ws/src/Thorvald/thorvald_vox_nav/maps/container_office_map.pcd # Provide a PCD format map
    # pcd_map_filename: /data/nextcloud_uni_bremen/slam_results/atlas_single_new_loop_candidates/map.pcd # Provide a PCD format map
    pointcloud_topic: "mrg_slam/map_points_service"                                  # If you are using a SLAM method that publishes pointclouds, you can use this topic

    # PCD PREPROCESS PARAMS
    pcd_map_downsample_voxel_size: -1.0                                       # Set to smaller if you do not want downsample pointclouds of the map
    pcd_map_transform:                                                        # Apply an OPTIONAL rigid-body transrom to pcd file, leave to all zeros if not wished
      translation:                                                            # Unit is meters
        x: 0.0
        y: 0.0
        z: 0.0
      rotation:                                                               # intrinsic rotation X-Y-Z (r-p-y)sequence, if you acquired map in camera frmae and want to represent
                                                                              # it in base_link or lidar frmae, you can specify the rotation here, UNit is Radians
        r: 0.0 #3.14
        p: 0.0 #1.57
        y: 0.0 #1.57
    apply_filters: True                                                      # Optional noise removal steps to apply to map
    remove_outlier_mean_K: 50                                                 # you can set it True and play with parameters if the map is noisy
    remove_outlier_stddev_threshold: 0.06
    remove_outlier_radius_search: 0.6
    remove_outlier_min_neighbors_in_radius: 5
    # COST REGRESSION CRITICS AND PARAMS
    uniform_sample_radius: 0.2
    surfel_radius: 0.6                                                       # Works as resolution of cost regression onto map
    max_allowed_tilt: 0.15                                                   # 1st Cost critic Any angle(radians) higher than this is marked as NON-traversable
    max_allowed_point_deviation: 0.20                                        # 2nd Cost critic Point deviation from plane, this could be viewed as roughness of each cell 
    max_allowed_energy_gap: 0.5                                              # 3rd Cost critic Max Energy in each cell, this is detemined by max height differnce between edge points of cell
    node_elevation_distance: 1.2                                             # According to cell_radius, cell centers are sampled from original point cloud map, they are elevated from the original cloud
    plane_fit_threshold: 0.1                                                 # when fitting a plane to each cell, a plane_fit_threshold is considered from plane fitting utility of PCL
    robot_mass: 0.1                                                          # approximate robot mass considering cell_radius, this isnt so important
    average_speed: 1.0                                                       # average robot speed(m/s) when calcuating kinetic energy m = 0.5 * (m * pow(v,2))
    cost_critic_weights: [0.45, 0.45, 0.1]                                   # Give weight to each cost critic when calculating final cost, see above 3 Cost Critic descriptions
    # PCD MAP IS CONVERTED TO OCTOMAP, THIS OCTOMAP IS THEN USED BY PLANNERS FOR
    # COLLISION CHECKING
    octomap_voxel_size: 0.4                                                  # determines resolution of Octomap
    octomap_publish_frequency: 1                                             # Used to determine publish frequncy of octomap visuals(pointclouds and markers)
    publish_octomap_visuals: true
    octomap_point_cloud_publish_topic: "vox_nav/map_server/octomap_pointcloud"                  # sensor_msgs::msg::PoinCloud2 that represents octomap
    octomap_markers_publish_topic: "vox_nav/map_server/octomap_markers"                         # visualization_msgs::msg::MarkeArray that represents octomap
    non_traversable_pointcloud_publish_topic: "vox_nav/map_server/non_traversable_pointcloud"   # sensor_msgs::msg::PoinCloud2 that represents non-traversable points
    traversable_pointcloud_publish_topic: "vox_nav/map_server/traversable_pointcloud"           # sensor_msgs::msg::PoinCloud2 that represents traversable points
    map_frame_id: "map"                                                      # This should be consistent with Gloabl EKF node , in robot_localization
    # utm_frame_id: "utm"
    # map_datum:                                                               # Datum coordinates of map is used to geo-reference the map 
    #                                                                          # If you are building a map with a SLAM method, you need to get a reading from GPS and absolute headed IMU
    #                                                                          # just before you start to build the map. See for example 
    #   latitude: 59.66424233333333         # CONTAINER OFFICE
    #   longitude: 10.76279866666667
    #   altitude: 0.9

    #   #latitude: 59.66597883333333          # ROBOT LAB
    #   #longitude: 10.777659
    #   #altitude: 111.0

    #   #latitude: 59.664226426330366          # UNEVEN GAZEBO
    #   #longitude: 10.762822144413834
    #   #altitude: 0.0
    #   quaternion:
    #     x: 0.0
    #     y: 0.0
    #     z: 0.0
    #     w: 1.0